R/plumber.R

```diff
@@ -322,7 +322,8 @@ plumber <- R6Class(
       httpuv::runServer(host, port, self)
     },
     mount = function(path, router){
-      path <- sub("([^/])$", "\\1/", path)
+      # Ensure that the path has both a leading and trailing slash.
+      path <- sub("^/?(.*)([^/])$", "/\\1\\2/", path)
```

Comment:
This won't catch the case where `path` has a trailing slash but no leading slash.

```R
sub("^/?(.*)([^/])$", "/\\1\\2/", "abc/")
#> [1] "abc/"
```

In general, I get a little nervous when there's conditional logic in a regexp, so I would try to avoid it if possible, especially if there are multiple conditions you're testing for. I'd suggest using two separate `if` statements, with something like this: 

```R
if (substr(path, 1, 1) != "/") {
  path <- paste0("/", path)
}
if (substr(path, nchar(path), nchar(path)) != "/") {
  path <- paste0(path, "/")
}
```

It's very verbose, but at least it's clear what it's doing.

Reply:
I prefer `substr` or even `startsWith()` as well, I just stuck to a regex to minimize changes.

> This won't catch the case where path has a trailing slash but no leading slash.

Ah. I thought I had tested that, but clearly not.

Reply:
+1 to `startsWith`

`?base::startsWith`
```
‘startsWith()’ is equivalent to but much faster than

     
     substring(x, 1, nchar(prefix)) == prefix
```

There's also an `endsWith`. TIL